cpp/dolfinx/fem/interpolate.h

```diff
@@ -107,8 +147,495 @@ void interpolate_from_any(Function<T>& u, const Function<T>& v)
   assert(v.function_space()->mesh());
   if (mesh->id() != v.function_space()->mesh()->id())
   {
-    throw std::runtime_error(
-        "Interpolation on different meshes not supported (yet).");
+    const int nProcs = dolfinx::MPI::size(MPI_COMM_WORLD);
+    const auto mpi_rank = dolfinx::MPI::rank(MPI_COMM_WORLD);
+
+    const int tdim = u.function_space()->mesh()->topology().dim();
+    const auto cell_map
+        = u.function_space()->mesh()->topology().index_map(tdim);
+    const int num_cells = cell_map->size_local() + cell_map->num_ghosts();
+    std::vector<std::int32_t> cells(num_cells, 0);
+    std::iota(cells.begin(), cells.end(), 0);
+
+    // Collect all the points at which values are needed to define the
+    // interpolating function
+    const xt::xtensor<double, 2> x = fem::interpolation_coords(
+        *u.function_space()->element(), *u.function_space()->mesh(), cells);
+
+    // This transposition is a quick and dirty solution and should be avoided
+    auto x_t = xt::zeros_like(xt::transpose(x));
+    for (decltype(x.shape(1)) i = 0; i < x.shape(1); ++i)
+    {
+      for (decltype(x.shape(0)) j = 0; j < x.shape(0); ++j)
+      {
+        x_t(i, j) = x(j, i);
+      }
+    }
+
+    dolfinx::geometry::BoundingBoxTree bb(*v.function_space()->mesh(), tdim,
+                                          0.0001);
+    auto globalBB = bb.create_global_tree(MPI_COMM_WORLD);
+
+    std::vector<std::vector<int>> candidates(x_t.shape(0));
+    for (decltype(x_t.shape(0)) i = 0; i < x_t.shape(0); ++i)
+    {
+      const auto xp = x_t(i, 0);
+      const auto yp = x_t(i, 1);
+      const auto zp = x_t(i, 2);
+      candidates[i]
+          = dolfinx::geometry::compute_collisions(globalBB, {xp, yp, zp});
+    }
+
+    std::vector<std::vector<double>> pointsToSend(nProcs);
+    for (decltype(candidates.size()) i = 0; i < candidates.size(); ++i)
+    {
+      const auto xp = x_t(i, 0);
+      const auto yp = x_t(i, 1);
+      const auto zp = x_t(i, 2);
+
+      for (const auto& p : candidates[i])
+      {
+        pointsToSend[p].push_back(xp);
```

Comment:
Using the insert function, as in: https://github.com/FEniCS/dolfinx/blob/3f8fa273315b904a9f394ea3d7072e7b940997fd/cpp/dolfinx/la/SparsityPattern.cpp#L59
i.e.
```
auto point = xt::row(x_t, i);
pointsToSend[p].insert(pointsToSend[p].end(), point.begin(), point.end())
```